client/my-sites/plans/jetpack-plans/utils.ts

```diff
@@ -517,11 +516,9 @@ export function buildCardFeatureItemFromFeatureKey(
 			text: feature.getTitle( variation ),
 			description: options?.withoutDescription ? undefined : feature.getDescription?.(),
 			subitems: subFeaturesKeys
-				? compact(
-						subFeaturesKeys.map( ( f ) =>
-							buildCardFeatureItemFromFeatureKey( f, options, variation )
-						)
-				  )
+				? subFeaturesKeys
+						.map( ( f ) => buildCardFeatureItemFromFeatureKey( f, options, variation ) )
+						.filter( ( f ): f is SelectorProductFeaturesItem => !! f )
```

Comment:
`filter( Boolean )` works like a charm in these instances:

```suggestion
						.filter( Boolean )
```